src/evo/cbtx.cpp

```diff
@@ -331,13 +331,26 @@ bool CheckCbTxBestChainlock(const CBlock& block, const CBlockIndex* pindex,
         return true;
     }
 
+    static Mutex cached_mutex;
+    static const CBlockIndex* cached_pindex GUARDED_BY(cached_mutex){nullptr};
+    static std::optional<std::pair<CBLSSignature, uint32_t>> cached_chainlock GUARDED_BY(cached_mutex){std::nullopt};
```

Comment:
Do we _really_ need to guard these with an extra mutex? The only place `CheckCbTxBestChainlock()` is called from is `ProcessSpecialTxsInBlock()` and the latter requires `cs_main` to be locked.

Reply:
I'd probably prefer not explicitly guarding it behind cs_main. And a non-recursive mutex is so cheap. This way potentially we can reduce cs_main usage in future w/o issue here.

Reply:
Makes sense.